### PR TITLE
Add basic docker-swarm test

### PR DESCRIPTION
**Pull request by mike-nguyen**
This is a basic sanity test for docker swarm and is not intended to
test all features of docker swarm.  The test uses the docker.io/httpd
container to validate the creation and scaling of services in a docker
swarm.

### CODE REVIEW

File: tests/docker-swarm/main.yml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+  tasks:
+    - name: Verify docker swarm is not running
+      command: docker info
+      register: di
+
+    - name: Fail if docker swarm is already running
+      fail:
+        msg: "Docker swarm is already running"
+      when:  "'Swarm: active' in di.stdout"
```
Comment (mike-nguyen) [inline tests/docker-swarm/main.yml]: I was looking for something similar to rpm-ostree status --json.  Maybe in the next few releases!

File: roles/docker_version_check/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+# vim: set ft=ansible:
+#
```
Comment (mike-nguyen) [inline roles/docker_version_check/tasks/main.yml]: Nice!  I can think of a few places where we can use this.

**Comment by jlebon** [comment]
👍

File: tests/docker-swarm/main.yml [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+    - name: Initialize docker swarm
+      command: docker swarm init --advertise-addr 127.0.0.1
+      register: dsi
```
Comment (miabbott) [inline tests/docker-swarm/main.yml]: nit - this doesn't appear to be used anywhere else

File: tests/docker-swarm/main.yml [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+      register: dsls
+
+    - name: Fail if {{ g_service_name }} is still in docker service ls output
+      fail:
+        msg: "{{ g_service_name }} should not be in docker service ls output"
+      when: g_service_name in dsls.stdout
+
+    - name: Deactivate swarm
+      command: docker swarm leave --force
+      register: dsl
```
Comment (miabbott) [inline tests/docker-swarm/main.yml]: nit - another variable that doesn't appear to be used elsewhere

**Comment by miabbott** [comment]
Two minor nits, otherwise LGTM

**Comment by mike-nguyen** [comment]
@miabbott fixed the two nits

**Comment by miabbott** [comment]
Looks good!  I ran it against F25AH + CAHC successfully.

**Pull request merged**